packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -436,6 +436,39 @@ const resolveObjectPatternValueNode = (
   return objectPatternValueNode;
 };
 
+/**
+ * Will return property key if it is different from its value and its value is
+ * equal to reference name for which we have to resolve binding.
+ *
+ * Eg. If we have something like `const { key: value } = { key: 'something' }`, and
+ * reference name is `value`, it will return `key` so that it can be resolved to
+ * 'something' otherwise it won't get resolved.
+ *
+ * @param node Object pattern node which we have to investigate
+ * @param referenceName Reference name for which `binding` to be resolved
+ */
+const getObjectPatternKeyWhenNotSameAsValue = (
```

Comment:
is there a less wordy name we could give this funtion?

`getDestructuredObjectPatternKey`? 🤔